internal/impl/confluent/processor_schema_registry_encode.go

```diff
@@ -348,26 +348,33 @@ func (s *schemaRegistryEncoder) getLatestEncoder(subject string) (schemaEncoder,
 	}
 
 	var codec *goavro.Codec
-	if codec, err = goavro.NewCodecForStandardJSON(resPayload.Schema); err != nil {
-		s.logger.Errorf("failed to parse response for schema subject '%v': %v", subject, err)
-		return nil, 0, err
+	if s.avroRawJSON {
+		if codec, err = goavro.NewCodecForStandardJSON(resPayload.Schema); err != nil {
+			s.logger.Errorf("failed to parse response for schema subject '%v': %v", subject, err)
+			return nil, 0, err
+		}
+	} else {
+		if codec, err = goavro.NewCodec(resPayload.Schema); err != nil {
+			s.logger.Errorf("failed to parse response for schema subject '%v': %v", subject, err)
+			return nil, 0, err
+		}
 	}
 
+	s.logger.Tracef("codec is: %#v", codec)
+
 	return func(m *service.Message) error {
-		var datum interface{}
-		if s.avroRawJSON {
-			b, err := m.AsBytes()
-			if err != nil {
-				return err
-			}
+		b, err := m.AsBytes()
+		if err != nil {
+			return err
+		}
 
-			if datum, _, err = codec.NativeFromTextual(b); err != nil {
-				return err
-			}
-		} else if datum, err = m.AsStructured(); err != nil {
+		datum, _, err := codec.NativeFromTextual(b)
+		if err != nil {
 			return err
 		}
```

Comment:
Hey @ekeric13, when I looked at this [on Discord](https://discord.com/channels/746368194196799589/754099840354615296/1006945611578351667), I got the impression that we still need this if / else logic here, but we should probably come up with some relevant test cases for both cases. Feel free to hit me up on Discord if you'd like to brainstorm through the corner cases.